annotations.md

```diff
@@ -28,7 +28,16 @@ This specification defines the following annotation keys, intended for but not l
 * **org.opencontainers.image.revision** Source control revision identifier for the packaged software.
 * **org.opencontainers.image.vendor** Name of the distributing entity, organization or individual.
 * **org.opencontainers.image.licenses** License(s) under which contained software is distributed as a [SPDX License Expression][spdx-license-expression].
-* **org.opencontainers.image.ref.name** Name of the reference for a target (string). SHOULD only be considered valid when on descriptors on `index.json` within [image layout](image-layout.md).
+* **org.opencontainers.image.ref.name** Name of the reference for a target (string).
+  * SHOULD only be considered valid when on descriptors on `index.json` within [image layout](image-layout.md).
+  * Character set of the value SHOULD conform to alphanum of `A-Za-z0-9` and separator set of `-._:@/+`
+  * An EBNF + regular expression like:
```

Comment:
Ah, reason for using `|` for alternates is probably so we can use `/…/` for marking regexps.  Since regexps are part of neither ABNF nor EBNF, I'd rather just drop the regexps.  I think:

```ABNF
ref = component *("/" component)
component = alphanum *(separator alphanum)
alphanum = 1*(ALPHA DIGIT)
separator = "-" / "." / "_" / ":" / "@" / "+" / "--"
```

(which leans on the [core rules][1] for `ALPHA` and `DIGIT`) is about as readable as the EBNF-ish + regexp syntax you're using in 5d36c514, and it has the additional benefit of using a [vanilla syntax][2].

[1]: https://tools.ietf.org/html/rfc5234#appendix-B.1
[2]: https://tools.ietf.org/html/rfc5234